api/v1beta1/machineset_types.go

```diff
@@ -129,6 +134,27 @@ const (
 	OldestMachineSetDeletePolicy MachineSetDeletePolicy = "Oldest"
 )
 
+// ANCHOR: RemediationStrategy
+
+// RemediationStrategy allows to define how control plane machine remediation happens.
+type RemediationStrategy struct {
+	// MaxInFlight determines how many in flight remediations should the MachineSet issue.
```

Comment:
I'm a bit concern about the impact on UX given the overlap with the MHC short circuiting semantics https://github.com/kubernetes-sigs/cluster-api/blob/main/api/v1beta1/machinehealthcheck_types.go#L53-L65

Let's be specific about the value this new field would add, the use cases it supports better by complementing the MHC builtin API and how we want to communicate it clearly for consumers.

Reply:
MaxUnhealthy and UnhealthyRange don't have any impact whatsoever on the amount of inflight remediation, it just controls if to stop making machines for remediation.

This PR adds a different knob in the same way that KubeadmControlPlane has RemediationStrategy

Reply:
Agreed. It effectively have an impact on the number of machines that can be remediated at time though.
I think we'll want to have good docs/examples to articulate how they play together. I can see this as a common question from consumers less familiar with the APIs.
https://cluster-api.sigs.k8s.io/tasks/automated-machine-management/healthchecking#creating-a-machinehealthcheck
https://cluster-api.sigs.k8s.io/tasks/automated-machine-management/healthchecking#remediation-short-circuiting

Reply:
May be at least mention the built in remediation capabilities section in https://cluster-api.sigs.k8s.io/tasks/automated-machine-management/healthchecking#limitations-and-caveats-of-a-machinehealthcheck

Reply:
Actually opened an issue to deprecate those fields in #10722